`npm test` => all tests fail

```
> haiku9@1.1.0-beta-14 test /vagrant/haiku9
> coffee test/index.coffee

 Haiku9 static-site generation
   Compile Stylus files AssertionError: File site.css not built
 Haiku9 static-site generation
   Passes through json files AssertionError: File manifest.json not built
 Haiku9 static-site generation
   Compiles Jade files AssertionError: File index.html not built
 Haiku9 static-site generation
   Compiles CoffeeScript files AssertionError: File site.js not built
 Haiku9 serve task
   Starts a local server on the default port AssertionError: Server did not respond
 Haiku9 static-site generation
   Passes through xml files AssertionError: File browserconfig.xml not built
Beginning task 'build'...
Beginning task 'survey'...
Beginning task 'survey/pass-thru'...
Beginning task 'build'...
Beginning task 'survey'...
Beginning task 'survey/pass-thru'...
Task 'survey/pass-thru' completed in 394ms.
Beginning task 'build'...
Beginning task 'survey/image'...
Beginning task 'survey'...
Beginning task 'survey/pass-thru'...
Task 'survey/image' completed in 6ms.
Beginning task 'survey/stylus'...
Task 'survey/stylus' completed in 5ms.
Beginning task 'survey/jade'...
Beginning task 'data'...
Task 'data' completed in 32ms.
Task 'survey/jade' completed in 40ms.
Beginning task 'survey/coffee'...
Task 'survey/coffee' completed in 3ms.
Beginning task 'survey/markdown'...
Beginning task 'build'...
Task 'survey/markdown' completed in 2ms.
Beginning task 'survey'...
Beginning task 'survey/posts'...
Beginning task 'survey/pass-thru'...
Task 'survey/posts' completed in 1ms.
Beginning task 'survey/bundle'...
Task 'survey/bundle' completed in 5ms.
Task 'survey' completed in 465ms.
Task 'build' completed in 484ms.
Task 'survey/pass-thru' completed in 182ms.
Beginning task 'survey/image'...
Task 'survey/image' completed in 8ms.
Beginning task 'survey/stylus'...
Task 'survey/stylus' completed in 2ms.
Beginning task 'survey/jade'...
Beginning task 'data'...
Beginning task 'build'...
Beginning task 'survey'...
Beginning task 'survey/pass-thru'...
Task 'data' completed in 14ms.
Task 'survey/jade' completed in 15ms.
Beginning task 'survey/coffee'...
Task 'survey/coffee' completed in 1ms.
Beginning task 'survey/markdown'...
Task 'survey/markdown' completed in 1ms.
Beginning task 'survey/posts'...
Task 'survey/posts' completed in 1ms.
Beginning task 'survey/bundle'...
Task 'survey/bundle' completed in 7ms.
Task 'survey' completed in 218ms.
Task 'build' completed in 229ms.
Task 'survey/pass-thru' completed in 200ms.
Beginning task 'survey/image'...
Task 'survey/image' completed in 6ms.
Beginning task 'survey/stylus'...
Task 'survey/stylus' completed in 2ms.
Beginning task 'survey/jade'...
Beginning task 'data'...
Task 'data' completed in 10ms.
Task 'survey/jade' completed in 13ms.
Beginning task 'survey/coffee'...
Task 'survey/coffee' completed in 1ms.
Beginning task 'survey/markdown'...
Task 'survey/markdown' completed in 1ms.
Beginning task 'survey/posts'...
Task 'survey/posts' completed in 1ms.
Beginning task 'survey/bundle'...
Task 'survey/bundle' completed in 2ms.
Task 'survey' completed in 228ms.
Task 'build' completed in 237ms.
Task 'survey/pass-thru' completed in 220ms.
Beginning task 'survey/image'...
Beginning task 'build'...
Beginning task 'survey'...
Beginning task 'survey/pass-thru'...
Task 'survey/image' completed in 9ms.
Beginning task 'survey/stylus'...
Task 'survey/stylus' completed in 0ms.
Beginning task 'survey/jade'...
Beginning task 'data'...
Task 'data' completed in 5ms.
Task 'survey/jade' completed in 6ms.
Beginning task 'survey/coffee'...
Task 'survey/coffee' completed in 1ms.
Beginning task 'survey/markdown'...
Task 'survey/markdown' completed in 2ms.
Beginning task 'survey/posts'...
Task 'survey/posts' completed in 0ms.
Beginning task 'survey/bundle'...
Task 'survey/bundle' completed in 4ms.
Task 'survey' completed in 252ms.
Task 'survey/pass-thru' completed in 156ms.
Beginning task 'survey/image'...
Task 'build' completed in 259ms.
Task 'survey/image' completed in 2ms.
Beginning task 'survey/stylus'...
Task 'survey/stylus' completed in 2ms.
Beginning task 'survey/jade'...
Beginning task 'data'...
Task 'data' completed in 5ms.
Task 'survey/jade' completed in 8ms.
Beginning task 'survey/coffee'...
Task 'survey/coffee' completed in 1ms.
Beginning task 'survey/markdown'...
Task 'survey/markdown' completed in 1ms.
Beginning task 'survey/posts'...
Task 'survey/posts' completed in 0ms.
Beginning task 'survey/bundle'...
Task 'survey/bundle' completed in 1ms.
Task 'survey' completed in 174ms.
Task 'build' completed in 183ms.
Task 'survey/pass-thru' completed in 80ms.
Beginning task 'survey/image'...
Task 'survey/image' completed in 2ms.
Beginning task 'survey/stylus'...
Task 'survey/stylus' completed in 1ms.
Beginning task 'survey/jade'...
Beginning task 'data'...
Task 'data' completed in 3ms.
Task 'survey/jade' completed in 4ms.
Beginning task 'survey/coffee'...
Task 'survey/coffee' completed in 1ms.
Beginning task 'survey/markdown'...
Task 'survey/markdown' completed in 1ms.
Beginning task 'survey/posts'...
Task 'survey/posts' completed in 0ms.
Beginning task 'survey/bundle'...
Task 'survey/bundle' completed in 1ms.
Task 'survey' completed in 91ms.
Task 'build' completed in 97ms.
```

--- a/test/helpers/build_helpers.coffee
+++ b/test/helpers/build_helpers.coffee
@@ -3,13 +3,11 @@ require "../../src/index"
 {join} = require "path"
 assert = require "assert"
 childProcess = require 'child_process'
-{shell, sleep, call, exists, chdir} = require "fairmont"
+{shell, call, exists, chdir, promise} = require "fairmont"
 
 HOME_DIR = join __dirname, "..", ".."
 BASE_FIXTURE_DIR = join HOME_DIR, "test", "fixtures"
 BUILD_FILE = join __dirname, "builder.coffee"
-
-BUILD_TIMEOUT = 5000
 
 fixtureDir = (fixture) ->
   join BASE_FIXTURE_DIR, fixture
@@ -28,7 +26,9 @@ build = (fixture) ->
     moveTo fixture
     yield shell "rm -rf build"
     moveTo fixture
-    childProcess.fork(BUILD_FILE)
+    yield promise (resolve, reject) ->
+      child = childProcess.fork(BUILD_FILE)
+      child.on 'exit', -> resolve()
     moveHome()
 
 # Predicate, returns true if the `expectedFile` exists
@@ -39,18 +39,7 @@ isBuilt = (fixture, expectedFile) ->
 
 # Asserts that the given test file exists under the fixture's build directory
 assertBuilt = (fixture, expectedFile) ->
-  call ->
-    start = now = new Date()
-    fileExists = false
-
-    while now - start <= BUILD_TIMEOUT
-      if fileExists = yield isBuilt(fixture, expectedFile)
-        break
-      else
-        yield sleep 10
-        now = new Date()
-
-    assert fileExists, "File #{expectedFile} not built"
+  assert yield isBuilt(fixture, expectedFile), "File #{expectedFile} not built"
 
 buildAndVerify = (fixture, expectedFile) ->
   call ->

